client/src/components/SignupForm.tsx

```diff
@@ -93,55 +92,49 @@ const SignupForm = () => {
           password: formValues.password,
           displayName: `${formValues.firstName} ${formValues.lastName}`,
         }),
-      });
-
-      const json = await registerRes.json();
-      if (import.meta.env.DEV) console.log(json);
-      localStorage.setItem("REFRESH_TOKEN", json.refreshToken);
-      localStorage.setItem("BEARER_TOKEN", json.token);
-
-      const getUserInfo = async () => {
-        try {
-          const response = await fetch(`http://localhost:4466/api/v1/users/${formValues.email}`);
-          const json = await response.json();
-          return json.user;
-        } catch (err) {
-          genericErrorHandler(err, toast);
-        }
-      };
-
-      const userData = await getUserInfo();
-      localStorage.setItem(
-        "USER",
-        JSON.stringify({
-          email: userData.email,
-          displayName: userData.displayName,
-          id: userData.id,
-          role: userData.role,
-        })
-      );
-
-      // Add user information to localStorage to hide edit/delete btn for posts/comments that do
-      // not belong to the user
-
-      if (registerRes.ok) {
-        navigate("/");
-        toast({
-          title: "Account created",
-          description: "Your account has been created successfully.",
-          status: "success",
-          duration: 10000,
-          isClosable: true,
+      }).then((response) => {
+        response.json().then((data) => {
+          localStorage.setItem("REFRESH_TOKEN", data.refreshToken);
+          localStorage.setItem("BEARER_TOKEN", data.token);
+          localStorage.setItem("USER", JSON.stringify(data.user));
+          if (response.ok) {
+            navigate("/");
+            toast({
+              title: "Account created",
+              description: "Your account has been created successfully.",
+              status: "success",
+              duration: 10000,
+              isClosable: true,
+            });
+          } else {
+            toast({
+              title: "Error",
+              description: "Error",
```

Comment:
should be `data.message` here I guess, or you can do `data.message || "Server error"` in case you are not sure